src/include/btree_inline.h

```diff
@@ -1793,41 +1793,20 @@ static inline bool
 __wt_page_evict_retry(WT_SESSION_IMPL *session, WT_PAGE *page)
 {
     WT_PAGE_MODIFY *mod;
-    WT_TXN_GLOBAL *txn_global;
-    wt_timestamp_t pinned_ts;
 
-    txn_global = &S2C(session)->txn_global;
-
-    /*
-     * If the page hasn't been through one round of update/restore, give it a try.
-     */
-    if ((mod = page->modify) == NULL || !FLD_ISSET(mod->restore_state, WT_PAGE_RS_RESTORED))
-        return (true);
+    mod = page->modify;
 
-    /*
-     * Retry if a reasonable amount of eviction time has passed, the choice of 5 eviction passes as
-     * a reasonable amount of time is currently pretty arbitrary.
-     */
-    if (__wt_cache_aggressive(session) ||
-      mod->last_evict_pass_gen + 5 < S2C(session)->cache->evict_pass_gen)
+    if (mod == NULL)
         return (true);
 
-    /* Retry if the global transaction state has moved forward. */
-    if (txn_global->current == txn_global->oldest_id ||
-      mod->last_eviction_id != __wt_txn_oldest_id(session))
+    if (__wt_cache_aggressive(session))
         return (true);
 
     /*
-     * It is possible that we have not started using the timestamps just yet. So, check for the last
-     * time we evicted only if there is a timestamp set.
+     * Retry if a reasonable amount of eviction time has passed, the choice of 5 eviction passes as
+     * a reasonable amount of time is currently pretty arbitrary.
      */
-    if (mod->last_eviction_timestamp != WT_TS_NONE) {
```

Comment:
Here it says if last eviction timestamp is the pinned timestamp, we will not retry. This is also blocking the eviction to add the page to the queue.